addons/web_editor/static/src/js/editor/snippets.editor.js

```diff
@@ -312,7 +312,10 @@ var SnippetEditor = Widget.extend({
         for (var i in this.styles) {
             await this.styles[i].onBuilt();
         }
+        // TODO In master differentiate device-based visibility.
+        this.options.wysiwyg._toggleVisibilityStatusIgnoreDeviceVisibility = true;
```

Comment:
I know this is meant as a temporary hacky fix but you could just use a global variable in this file instead of writing on `this.options.wysiwyg`, no? I let @detrouxdev handle the rest :pray: